woocommerce/woocommerce-ios
# Order Details: update Products, Tracking, and Add Tracking sections based on shipping labels

Part of #2167 

## Why

Now that the shipping label sections also show order items (products/variations) in an order, we only want to show the order items & quantity that are not yet included in a non-refunded shipping label so that the store owner can still create a shipping label for the remaining items in M2. Before this PR, the Products section shows the order items combined with refunded products. This PR updates the Products section to show the **order items combined with refunded products and non-refunded shipping labels**. It also hides the tracking sections if there are non-empty non-refunded shipping labels.

## Changes

- Added a helper function to `AggregateDataHelper` to combine aggregate order items with those from non-refunded shipping labels with a test case
- Added a helper function to `AggregatedShippingLabelOrderItems` that returns an array of order items from all of the given non-refunded shipping labels
- Added an extension helper to a collection of `ShippingLabel` to return non-refunded shipping labels
- In `OrderDetailsDataSource`:
  - Updates the Products section to show the combined aggregate order items from refunded products **plus non-refunded shipping labels**
  - Hides the tracking sections if there are non-empty non-refunded shipping labels
- Implemented `GeneratedCopiable` for `AggregateOrderItem` and ran `rake generate` after removing `TestKit` to generate `Models+Copiable.generated.swift` in WooCommerce target. This is used in `AggregateDataHelper` and the test case

## Testing

Prerequisites: the site has [WooCommerce Shipping & Tax plugin](https://docs.woocommerce.com/document/woocommerce-shipping-and-tax/) and there is an order with at least one non-refunded shipping label.

You can set up your store to use a test credit card with instructions in p91TBi-3xD-p2.

There are many edge cases like with refunded products and refunded shipping labels in multi-package scenario. Feel free to test what makes sense to you!

- Go to the orders tab
- Tap on an order with at least one non-refunded shipping label --> after syncing, the Products section should only show the products that aren't covered by the non-refunded shipping labels (the section is hidden if there are no products to show). Also, the Tracking and Add Tracking sections should be hidden

## Example screenshots

\ | before | after
-- | -- | --
there are remaining order items | ![Simulator Screen Shot - iPhone 11 - 2020-12-02 at 11 56 53](https://user-images.githubusercontent.com/1945542/100827225-38a55f00-3497-11eb-87fc-06ddb7c51954.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-02 at 11 54 12](https://user-images.githubusercontent.com/1945542/100827451-b4071080-3497-11eb-882a-f2c332e91b86.png)
no remaining order items | ![Simulator Screen Shot - iPhone 11 - 2020-12-02 at 11 57 04](https://user-images.githubusercontent.com/1945542/100827228-3cd17c80-3497-11eb-9bb8-cd47348e8712.png) | ![Simulator Screen Shot - iPhone 11 - 2020-12-02 at 11 54 36](https://user-images.githubusercontent.com/1945542/100827463-b8cbc480-3497-11eb-9650-a62a2f3b694f.png)

Update release notes:
